src/oops/SOLID/openClosePrinciple/before/taxes/TaxCalculatorFactory.java

```diff
@@ -0,0 +1,31 @@
+package oops.SOLID.openClosePrinciple.before.taxes;
+
+import oops.SOLID.openClosePrinciple.before.employees.Employee;
+import oops.SOLID.openClosePrinciple.before.employees.FullTimeEmployee;
+import oops.SOLID.openClosePrinciple.before.employees.Intern;
+import oops.SOLID.openClosePrinciple.before.employees.PartTimeEmployee;
+import oops.SOLID.openClosePrinciple.before.taxes.impl.FullTimeEmployeeTaxCalculator;
+import oops.SOLID.openClosePrinciple.before.taxes.impl.InternTaxCalculator;
+import oops.SOLID.openClosePrinciple.before.taxes.impl.PartTimeEmployeeTaxCalculator;
+
+
+public class TaxCalculatorFactory {
+
+    private static final FullTimeEmployeeTaxCalculator fullTimeEmployeeTaxCalculator = new FullTimeEmployeeTaxCalculator();
+    private static final PartTimeEmployeeTaxCalculator partTimeEmployeeTaxCalculator = new PartTimeEmployeeTaxCalculator();
+    private static final InternTaxCalculator internTaxCalculator = new InternTaxCalculator();
+
+    public static TaxCalculator getTaxCalculator(Employee e) {
+        if (e instanceof FullTimeEmployee)
```

Comment:
It would be better to replace this "if-else" block to fetch entries from a Map. This would also help in injecting the map from a bean.
For example, you can create and populate a map like `Map<Class<? extends Employee>, TaxCalculator>` and use it to fetch the respective `TaxCalculator`

Reply:
Agreed with @Kalyan7767. Same was discussed in the lecture.